llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -785,16 +785,24 @@ class YkIRWriter {
     AttributeSet FnAttrs = Attrs.getFnAttrs();
     for (auto &Attr : FnAttrs) {
       // - `cold` can be ignored.
+      // - `memory` are memory effect hints, which can be ignored.
       // - `nounwind` has no consequences for us at the moment.
       // - `returnstwice` can be ignored.
       // - `willreturn` function returns or has UB.
       if (Attr.isEnumAttribute() &&
           ((Attr.getKindAsEnum() == Attribute::Cold) ||
+           (Attr.getKindAsEnum() == Attribute::Memory) ||
            (Attr.getKindAsEnum() == Attribute::NoUnwind) ||
            (Attr.getKindAsEnum() == Attribute::ReturnsTwice) ||
            (Attr.getKindAsEnum() == Attribute::WillReturn))) {
         continue;
       }
+      // Attributes can also be represented as strings, integers or "types"
+      // which won't be caught by the above. You can always convert an
+      // attribute to a string, so a string comparison always works.
+      if (Attr.getAsString().find("memory(") == 0) {
```

Comment:
I think I ran into this one too and used `Attr.hasAttribute(Attribute::Memory)` to catch this case. Might be better than a string comparison?